File: Assignment 05/Poland.py
arr = [int(x) for x in input().split()]

n = arr[0]
m = arr[1]
used = []
both = []
poland = []
enemy = []
for x in range(n):
    s = input()
    if (used.count(s)):
        continue
    poland.insert(0, s)
    used.insert(0, s)

for x in range(m):
    s = input ()
    if(used.count(s)):
        if (poland.count(s)):
            both.insert(0, s)
            poland.remove(s)
        else:
            used.insert(0, s) 
            enemy.insert(0, s)

psz = len(poland)
bsz = len(both)
esz = len(enemy)

temp = bsz/2
esz += temp
psz += (bsz-temp)
	
if psz>esz:
    print ("YES")
else:
    print("NO")
	

File: Assignment 05/Poland.cpp
#include<bits/stdc++.h>
#include<stdio.h>
#include<iostream>
#include<string>
#include<algorithm>
#include<cstdio>
#include<vector>

using namespace std;

int main()
{
	set<string> poland,enemy,common,done;
	char s[1005];
	int n, m;
	cin >> n >> m;
	
	for(int i = 0 ; i < n ; i++){
		cin >> s;
		
		if(done.count(s))
			continue ;
		poland.insert(s);
		done.insert(s);
	}
	
	for(int i = 0 ; i < m ; i++){
		cin >> s;
		if(done.count(s)){
			if(poland.count(s)){
				common.insert(s);
				poland.erase(s);
			}
		}
		else{
			done.insert(s);
			enemy.insert(s);
		}
	}
	
	int one = poland.size();
	int two = common.size();
	int three = enemy.size();

	int temp = two/2;
	three += temp;
	one += (two-temp);
	
	if (one > three){
        cout << "YES" << endl;
    }
    else
    {
        cout<< "NO" << endl;
    }
    
	return 0;
}
    